lib/cli-engine.js

```diff
@@ -300,18 +300,31 @@ function processFile(filename, configHelper, options) {
 
 /**
  * Returns result with warning by ignore settings
- * @param {string} filePath File path of checked code
- * @returns {Result} Result with single warning
+ * @param {string} filePath - File path of checked code
+ * @param {string} baseDir  - Absolute path of base directory
+ * @returns {Result}           Result with single warning
  * @private
  */
-function createIgnoreResult(filePath) {
+function createIgnoreResult(filePath, baseDir) {
+    var message;
+
+    if (/^\./.test(path.basename(filePath))) {
+        message = "Hidden file ignored by default.  Use '--ignore-pattern !.*' to override.";
```

Comment:
To be clear, I'm proposing to adapt it like this:

```
var relativePath = path.relative(baseDir, filePath));

if (/^\./.test(path.basename(filePath))) {
  message = "File ignored by default.  Use '--ignore-pattern \"!" + relativePath + "\"' to override.";
} else if (baseDir && /^node_modules/.test(relativePath) {
  message = "File ignored by default. Use '--ignore-pattern \"!node_modules/*\"' to override.";
} else if (baseDir && /^bower_components/.test(path.relative(baseDir, filePath))) {
  message = "File ignored by default. Use '--ignore-pattern \"!bower_components/*\"' to override.";
} else {
  message = "File ignored because of a matching ignore pattern. Use '--no-ignore' to override.";
}
```